### ISSUE TITLE
Provisioning the control VM failed in the Kubernetes cluster 

### ISSUE THREAD

**Issue by Tbaugus44** [apache/cloudstack#8140]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
cloud.kubernetes.service
~~~

##### CLOUDSTACK VERSION

~~~
4.18.1.0
~~~

##### CONFIGURATION

Advanced Networked Zone 

##### OS / ENVIRONMENT

Cloudstack 4.18.1.0 is running on bare-metal Ubuntu 22.04.03 LTS. KVM Hypervisor 

##### SUMMARY

When We try to create a Kubernetes Cluster we get the error Provisioning the control VM failed in the Kubernetes cluster. 4.14 release has you updated the cloud.kubernetes.cluster.template.name.kvm and you are able to create a cluster. 4.18.1 does not have this Global setting 

##### STEPS TO REPRODUCE

![image](https://github.com/apache/cloudstack/assets/148808916/2f431584-ccdc-4855-ba2b-e6cdb5722902)

~~~

~~~

##### EXPECTED RESULTS

~~~
Cloudstack to create a Kubernetes Cluster
~~~

##### ACTUAL RESULTS

Cloudstack failed to create a Kubernetes cluster 

~~~
Provisioning the control VM failed in the Kubernetes cluster 
~~~

**Comment by kiranchavala** [comment]
@Tbaugus44 could you please let me know the kubernetes version you are launching 

https://download.cloudstack.org/cks/


Also, make sure you set the global settings value of  "endpoint.url" to the management server IP

**Comment by shwstppr** [comment]
@Tbaugus44 global settings `cloud.kubernetes.cluster.template.name.*` have been removed as now k8s node deployment is done using the systemvm template and you do not need to register a separate template.
Based on the screenshot I guess you are using 1.27.3 version? cc @kiranchavala 
Can you please provide logs for the failure?

**Comment by Tbaugus44** [comment]
[management-server.log](https://github.com/apache/cloudstack/files/13165989/management-server.log)
Please look between 8:30 and 8:40 10/24/2023 

 I also checked the "endpoint.url" global setting and it shows the management servers IP.

**Comment by Tbaugus44** [comment]
![image](https://github.com/apache/cloudstack/assets/148808916/70d77c82-e497-4a38-b9b0-9206137144bb)
I'm also seeing this error now. this happens after I create the cluster and try to start it from an errored state.

**Comment by shwstppr** [comment]
@Tbaugus44 I'm seeing this error,
```
2023-10-25 08:36:12,671 ERROR [c.c.v.UserVmManagerImpl] (API-Job-Executor-2:ctx-e6e21680 job-26576 ctx-ec4c1914) (logid:448e7210) error during resource reservation and allocation
java.lang.NullPointerException                                                     
>---at com.cloud.vm.UserVmManagerImpl.validateRootDiskResize(UserVmManagerImpl.java:4642)
>---at com.cloud.vm.UserVmManagerImpl.commitUserVm(UserVmManagerImpl.java:4446)        
>---at com.cloud.vm.UserVmManagerImpl.commitUserVm(UserVmManagerImpl.java:4630)    
>---at com.cloud.vm.UserVmManagerImpl.getUncheckedUserVmResource(UserVmManagerImpl.java:4261)
>---at com.cloud.vm.UserVmManagerImpl.getCheckedUserVmResource(UserVmManagerImpl.java:3972)
>---at com.cloud.vm.UserVmManagerImpl.createVirtualMachine(UserVmManagerImpl.java:3955)
>---at com.cloud.vm.UserVmManagerImpl.createAdvancedVirtualMachine(UserVmManagerImpl.java:3729)
>---at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>---at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>---at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```

Can you please check the template used for deploying the cluster's node VM is in the correct state?
From UI check if it is showing size value in the template details? Are systemvms using same template?
Check entries for template in vm_template and template_store_ref tables if needed

**Comment by Tbaugus44** [comment]
this is what I see in the UI 
![image](https://github.com/apache/cloudstack/assets/148808916/e3de4b1f-e351-4e31-819f-71311322c139)
![image](https://github.com/apache/cloudstack/assets/148808916/557a0c15-3332-41b7-a6e1-0b0dd72076dc)

**Comment by shwstppr** [comment]
@Tbaugus44 is this an upgraded env?
I suspect vm_template table does not have a size value for your template.
What version of the template does systemvm view show?

**Comment by Tbaugus44** [comment]
this is what  SSVM shows. is there a way to re-download the systemvm template so that the disk value updates? this is a lab enviroment with plans on taking what we learned and starting a production env.
![image](https://github.com/apache/cloudstack/assets/148808916/ca500a64-2cad-481d-9877-ebb25c3c8ce9)

**Comment by Tbaugus44** [comment]
im getting this error now
![image](https://github.com/apache/cloudstack/assets/148808916/13c519b2-019c-40b1-914f-1940b1d8fdbb)

**Comment by Tbaugus44** [comment]
I wanted to Report a solution,

The issue was with the SystemVmTemplate. We downloaded a new template from Shapeblue since their repository is much faster than the Apachecloustack repo. http://packages.shapeblue.com/systemvmtemplate/ 

From there we were getting the above error " Failed to start Kubernetes Cluster: network cannot be started. We changed the Global setting router.template.kvm to the new system template. We also updated the template under the image tab, we set the new template to a system template. 

Kubernetes is now working in our environment after these changes.

Thanks!
